luigi/contrib/lsf.py

```diff
@@ -0,0 +1,354 @@
+
+The procedure:
+
+- Pickle the class
+- Construct a bsub argument that runs a generic runner function with the path to the pickled class
+- Runner function loads the class from pickle
+- Runner function hits the work button on it
+
+"""
+
+LOGGER = logging.getLogger('luigi-interface')
+
+
+def track_job(job_id):
+    """
+    Tracking is done by requesting each job and then searching for whether the job
+    has one of the following states:
+    - "RUN",
+    - "PEND",
+    - "SSUSP",
+    - "EXIT"
+    based on the LSF documentation
+    """
+    cmd = "bjobs -noheader -o stat %d" % job_id
+    track_job_proc = subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, shell=True)
+    status = track_job_proc.communicate()[0].strip('\n')
+    return status
+
+
+def kill_job(job_id):
+    """
+    Kill a running LSF job
+    """
+    subprocess.call(['bkill', job_id])
+
+
+class LSFJobTask(luigi.Task):
+    """
+    Takes care of uploading and executing an LSF job
+    """
+
+    n_cpu_flag = luigi.IntParameter(default=2, significant=False)
+    shared_tmp_dir = luigi.Parameter(default='/tmp', significant=False)
+    resource_flag = luigi.Parameter(default='mem=8192', significant=False)
+    memory_flag = luigi.Parameter(default='8192', significant=False)
+    queue_flag = luigi.Parameter(default='queue_name', significant=False)
+    runtime_flag = luigi.IntParameter(default=60)
+    job_name_flag = luigi.Parameter(default='')
+    poll_time = luigi.FloatParameter(
+        significant=False, default=5,
+        description="specify the wait time to poll bjobs for the job status")
+    save_job_info = luigi.BoolParameter(default=False)
+    output = luigi.Parameter(default='')
+    extra_bsub_args = luigi.Parameter(default='')
+
+    job_status = None
+
+    def fetch_task_failures(self):
+        """
+        Read in the error file from bsub
+        """
+        error_file = os.path.join(self.tmp_dir, "job.err")
+        if os.path.isfile(error_file):
+            with open(error_file, "r") as f_err:
+                errors = f_err.readlines()
+        else:
+            errors = ''
+        return errors
+
+    def fetch_task_output(self):
+        """
+        Read in the output file
+        """
+        # Read in the output file
+        with open(os.path.join(self.tmp_dir, "job.out"), "r") as f_out:
+            outputs = f_out.readlines()
+        return outputs
+
+    def _init_local(self):
+
+        base_tmp_dir = self.shared_tmp_dir
+
+        random_id = '%016x' % random.getrandbits(64)
+        task_name = random_id + self.task_id
+        # If any parameters are directories, if we don't
+        # replace the separators on *nix, it'll create a weird nested directory
+        task_name = task_name.replace("/", "::")
+
+        self.tmp_dir = os.path.join(base_tmp_dir, task_name)
+        # Max filename length
+        max_filename_length = os.fstatvfs(0).f_namemax
+        self.tmp_dir = self.tmp_dir[:max_filename_length]
```

Comment:
I'm not sure I follow what you're trying to do. I'm unsure if f_namemax is the max length of a filename component or the entire path. Either way, you're truncating the base_tmp_dir + task_name concatenated path to f__namemax then adding packages.tar a bit later. If f_namemax is the max total path length, wouldn't making it longer be an issue? If it's the max length of a component of the path, shouldn't it be truncated before prepending the base path?